Synview-backend/routes/github_routes.ts

```diff
@@ -0,0 +1,75 @@
+import { PrismaClient } from "../generated/prisma/client.ts";
+import { Session } from "https://deno.land/x/oak_sessions/mod.ts";
+import { withAccelerate } from "@prisma/extension-accelerate";
+import AuthMiddleware from "../middleware/auth_middleware.ts";
+import { Router } from "@oak/oak";
+import { GithubInfoSchema } from "../../common/schemas.ts";
+import { Octokit } from "npm:@octokit/rest";
+import { Page } from "https://deno.land/x/mongo@v0.31.2/src/utils/saslprep/memory_pager.ts";
+const env = Deno.env.toObject();
+type AppState = {
+  session: Session;
+};
+const GitHubToken = env.GITHUB_TOKEN;
+const octokit = new Octokit({
+  auth: GitHubToken,
+});
+const githubRouter = new Router<AppState>();
+const prisma = new PrismaClient().$extends(withAccelerate());
+
+githubRouter.use(AuthMiddleware);
+
+githubRouter
+  .post("/suncCommitsHook", async (context) => {})
+  .post("/syncCommits", async (context) => {
+    try {
+      const GithubInfo = GithubInfoSchema.parse(
+        await context.request.body.json()
+      );
+      let page = 1;
```

Comment:
Is this called from the frontend? If so, `page` should be a parameter passed in the endpoint, hardcoding it does not seem right. The `while` loop below could potentially go through a LOT of data

It is weird to see anything in programming that starts at 1, usually numbers like page would be zero-indexed and start at 0, is this right?

Reply:
Where is this endpoint called from?

Reply:
There are some repos with so many commits that it would almost certainly OOM (out-of-memory) crash any server that ran this code - usually, you'd send the page from the frontend and enable paging, ie. it loads a small batch and then can scroll

But getting the two tech challenges done is probably more important first